src/passes/Print.cpp

```diff
@@ -78,6 +78,16 @@ struct PrintSExpression : public Visitor<PrintSExpression> {
     return name;
   }
 
+  std::ostream& printName(Name name) {
+    // we need to quote names if they have tricky chars
```

Comment:
Yes, internal names (the ones beginning with `$`) can't be quoted in the current spec format.